psi4/driver/json_wrapper.py

```diff
@@ -50,9 +49,132 @@
+            json_data["raw_output"] = f.read()
+        os.unlink(outfile)
+
+    return json_data
+
+
+def run_json_qc_schema(json_data):
+    """
+    An implementation of the QC JSON Schema (molssi-qc-schema.readthedocs.io/en/latest/index.html#) implementation in Psi4.
+
+
+    Parameters
+    ----------
+    json_data : JSON
+        Please see molssi-qc-schema.readthedocs.io/en/latest/spec_components.html for further details.
+
+    Notes
+    -----
+    !Warning! This function is experimental and likely to change in the future.
+    Please report any suggestions or uses of this function on github.com/MolSSI/QC_JSON_Schema.
+
+    Examples
+    --------
+
+    """
+
+    # This is currently a forced override
+    json_data["schema_version"] = 0
+
+    json_data["provenance"] = {"creator": "QM Program", "version": psi4.__version__, "routine": "psi4.json.run_json"}
+
+    # QCDB, help Lori!
+    mol = core.Molecule.from_arrays(geom=json_data["molecule"]["geometry"], elem=json_data["molecule"]["symbols"], units="Bohr")
+
+    method = json_data["model"]["method"] + "/" + json_data["model"]["basis"]
+    val, wfn = methods_dict[json_data["driver"]](method, return_wfn=True, molecule=mol)
+
+
+    if json_data["driver"] == "energy":
+        json_data["return_result"] = val
+    elif json_data["driver"] in ["gradient", "hessian"]:
+        json_data["return_result"] = val.np.ravel().tolist()
+    else:
+        raise KeyError("Did not understand key %s")
+
+    # Pull out a standard set of SCF properties
+    psi_props = psi4.core.get_variables()
+    json_data["psi4:properties"] = psi_props
+
+    props = {
+        "calcinfo_nbasis": wfn.nso(),
+        "calcinfo_nmo": wfn.nmo(),
+        "calcinfo_nalpha": wfn.nalpha(),
+        "calcinfo_nbeta": wfn.nbeta(),
+        "calcinfo_natom": mol.geometry().shape[0],
+        "scf_one_electron_energy": psi_props["ONE-ELECTRON ENERGY"],
+        "scf_two_electron_energy": psi_props["TWO-ELECTRON ENERGY"],
+        "nuclear_repulsion_energy": psi_props["NUCLEAR REPULSION ENERGY"],
+        "scf_dipole_moment": [psi_props[x] for x in ["SCF DIPOLE X", "SCF DIPOLE Y", "SCF DIPOLE Z"]],
+        "scf_iterations": int(psi_props["SCF ITERATIONS"]),
+        "scf_total_energy": psi_props["SCF TOTAL ENERGY"],
+    }
+
+    # Pull out optional SCF keywords
+    other_scf = [("DFT VV10 ENERGY", "scf_vv10_energy"), ("DFT XC ENERGY", "scf_xc_energy"),
+                 ("DISPERSION CORRECTION ENERGY", "scf_dispersion_correction_energy")]
+    for pkey, skey in other_scf:
+        if (pkey in psi_props) and (psi_props[pkey] != 0):
+            props[skey] = psi_props[pkey]
+
+    # Write out MP2 keywords
+    if "MP2 CORRELATION ENERGY" in psi_props:
+        props["mp2_same_spin_correlation_energy"] = psi_props["MP2 SAME-SPIN CORRELATION ENERGY"]
+        props["mp2_opposite_spin_correlation_energy"] = psi_props["MP2 OPPOSITE-SPIN CORRELATION ENERGY"]
+        props["mp2_singles_energy"] = 0.0
```

Comment:
If you care about singles energy, I have some reform (https://github.com/loriab/psi4/commit/263c102469c1ed065badfa3069ea3d2f60c1a8c8). If not, it can wait.

Reply:
Eh, I think it can wait for now.

Reply:
Main difference here is energies are energies and will always be produced. For properties, what happens if someone asks for `energy('MP2')` and then several properties? You will get SCF properties which can lead to significant confusion. The `properties` fields are explicitly labeled for each level of theory while the `return_result` is simple what was requested.